api/envoy/type/matcher/metadata.proto

```diff
@@ -0,0 +1,73 @@
+syntax = "proto3";
+
+package envoy.type.matcher;
+option go_package = "matcher";
+
+import "envoy/type/matcher/string.proto";
+import "envoy/type/matcher/number.proto";
+
+import "validate/validate.proto";
+
+// [#protodoc-title: MetadataMatcher]
+
+// MetadataMatcher provides a general interface to check if a given value is matched in
+// :ref:`Metadata <envoy_api_msg_core.Metadata>`. It uses `filter` and `path` to retrieve the value
+// from the Metadata and then check if it's matched to one of the specified values.
+//
+// An example use of MetadataMatcher is specifying additional metadata in envoy.filters.http.rbac to
+// enforce access control based on dynamic metadata in a request.
+message MetadataMatcher {
+  // Specifies the segment in a path to retrieve value from Metadata.
+  // Note: Array is not supported for now and will result a not match directly.
+  message PathSegment {
+    oneof segment {
+      option (validate.required) = true;
+
+      // If specified, use the key to retrieve the value in a Struct.
+      string key = 1 [(validate.rules).string.min_bytes = 1];
+    }
+  }
+
+  // Specifies the value to match. Only primitive value is supported. For non-primitive value, the
```

Comment:
"are supported", "For non-primitive values,"

Reply:
Done